### PR TITLE
Add scan API to Wi-Fi Manager and its testcase

### PR DESCRIPTION
**Pull request by davidfather**
1. Add scan API to Wi-Fi Manager
Using this API, in station mode, can scan nearby access points and get scan results including ssid, bssid, and rssi.

2. Add unit testcase for scan API of Wi-Fi Manager
Test the basic functions of scan API

### CODE REVIEW

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+	NULL,	//wifi_sta_connected,
+	NULL,	//wifi_sta_disconnected,
+	NULL,	//wifi_softap_sta_joined,
+	NULL,	//wifi_softap_sta_left,
+	NULL,	// in station mode, this callback function is called when scanning ap is done.
+};
+
+static void wifi_sta_scan_ap_done(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res)
+{
+	printf("[pkes] T%d --> %s\n", getpid(), __FUNCTION__);
```
Comment (pillip8282) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: please remove pkes which is personal log.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+static void utc_wifi_manager_set_mode_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_set_mode(SOFTAP_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_FAIL);
+
+	wifi_manager_softap_config_s ap_config = {"PKES", 1, "1111122222"};
```
Comment (pillip8282) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: how about to set AP's configurations by Kconfig

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+	/* change to station mode */
+	ret = wifi_manager_set_mode(STA_MODE, NULL);
+	TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+
+	/* Connect to AP
+	 * You need to change configuration of ap which you actually try to connect to.
+	 * Otherwise, this function will fail.
+	 */
+	wifi_manager_ap_config_s config = {\
```
Comment (pillip8282) [inline apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: same as above

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 1 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: I remove it.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 2 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: I do change.

File: apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c (hunk 3 above)
Comment (davidfather) [reply apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c]: I do change

**Pull request merged**